### PR TITLE
feature: limit number of currencies/markets

### PR DESCRIPTION
**Pull request by dinesh-skyach**
feature: limit number of currencies/markets

### CODE REVIEW

File: app/models/currency.rb [hunk 1]
```diff
@@ -33,6 +33,11 @@ class Currency < ApplicationRecord
 
   # == Validations ==========================================================
 
+  validate on: :create do
+    errors.add(:max, 'Currency limit has reached')\
+      if Currency.count >= ENV['MAX_CURRENCIES'].to_i
```
Comment (calj) [inline app/models/currency.rb]: Please disable this check if the variable is not set.
And correct message: 'Currency limit has been reached'

File: app/models/market.rb [hunk 2]
```diff
@@ -64,6 +64,9 @@ class Market < ApplicationRecord
   end
 
   validate on: :create do
+    errors.add(:max, 'Market limit has reached')\
+      if Market.count >= ENV['MAX_MARKETS'].to_i
```
Comment (mnaichuk) [inline app/models/market.rb]: Better don't use one line `if` here with `\`

Comment (dinesh-skyach) [reply app/models/market.rb]: Ok sure, I will make the changes.

File: app/models/currency.rb (hunk 1 above)
Comment (dinesh-skyach) [reply app/models/currency.rb]: Ok sure, I have made these changes please review it once.

File: app/models/market.rb (hunk 2 above)
Comment (dinesh-skyach) [reply app/models/market.rb]: I have made some changes, please review it once.

File: spec/models/currency_spec.rb [hunk 3]
```diff
@@ -107,4 +107,15 @@
       expect(fake_currency.subunits).to eq(2)
     end
   end
+
+  context 'validate max currency' do
+    before { ENV['MAX_CURRENCIES'] = '6' }
+    after  { ENV['MAX_CURRENCIES'] = '10' }
```
Comment (mnaichuk) [inline spec/models/currency_spec.rb]: maybe better to just delete it from ENV? can be case that we will create more than 10 currencies during the test

Comment (dinesh-skyach) [reply spec/models/currency_spec.rb]: Ok sure, I will set `MAX_CURRENCIES` and `MAX_MARKETS` to nil.

Comment (dinesh-skyach) [reply spec/models/currency_spec.rb]: I have made the changes, please have a look.

**Review by calj — APPROVED** [review]

**Pull request merged**